react-components/src/data-providers/FdmSDK.ts

```diff
@@ -529,16 +530,32 @@ export class FdmSDK {
     throw new Error(`Failed to fetch instances. Status: ${result.status}`);
   }
 
-  public async queryNodesAndEdges<const T extends Query>(query: T): Promise<QueryResult<T>> {
-    const result = await this._sdk.post(this._queryEndpoint, { data: query });
-    if (result.status === 200) {
-      return { items: result.data.items, nextCursor: result.data.nextCursor };
+  public async queryAllNodesAndEdges<
+    TQueryRequest extends QueryRequest,
+    TypedSelectSources extends SelectSourceWithParams = SelectSourceWithParams
+  >(query: TQueryRequest): Promise<QueryResult<TQueryRequest, TypedSelectSources>> {
+    let result = await queryNodesAndEdges<TQueryRequest, TypedSelectSources>(query, this._sdk);
+    let items = result.items;
+    while (result.nextCursor !== undefined && Object.keys(result.nextCursor).length !== 0) {
+      const newQuery = { ...query, cursors: result.nextCursor };
+      result = await queryNodesAndEdges<TQueryRequest, TypedSelectSources>(newQuery, this._sdk);
+      items = mergeQueryResults(items, result.items);
     }
-    throw new Error(`Failed to fetch instances. Status: ${result.status}`);
+
+    return { items };
+  }
+
+  public async queryNodesAndEdges<
+    TQueryRequest extends QueryRequest,
+    TypedSelectSources extends SelectSourceWithParams = SelectSourceWithParams
+  >(query: TQueryRequest): Promise<QueryResult<TQueryRequest, TypedSelectSources>> {
+    return await queryNodesAndEdges<TQueryRequest, TypedSelectSources>(query, this._sdk);
```

Comment:
Although I dont see anything smelling with the queryNodesAndEdges and the code related, I think it would be good if Anders could pay some extra attention in here, including the QueryResult type definition inside the queryNodesAndEdges.ts

Reply:
This is taken directly from @christjt - but I'll ask Anders to look as well :sweat_smile: